test/built-ins/Temporal/Duration/prototype/with/copy-properties-not-undefined.js

```diff
@@ -0,0 +1,23 @@
+// Copyright (C) 2022 Igalia, S.L. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-temporal.duration.prototype.with
+description: PreparePartialTemporalFields copies only defined properties of source object
+info: |
+  4. For each value _property_ of _fieldNames_, do
+    a. Let _value_ be ? Get(_fields_, _property_).
+    b. If _value_ is not *undefined*, then
```

Comment:
Consider also testing an explicit `undefined` property

Reply:
I've added one extra `prop: undefined` to each property bag, @jessealama let me know what you think